design-toolkit/components/src/index.css

```diff
@@ -234,6 +234,14 @@
   --icon-color: --value(--color- *);
 }
 
+@custom-variant data-selected (&[data-selected~="true"]);
```

Comment:
There's already a variant for selected. If there's an issue with the existing one then lets discuss what needs to be fixed